CMakeLists.txt

```
@@ -24,3 +24,15 @@ target_include_directories(
         INTERFACE
         include/
 )
+
+if (UNIX)
+    set(COMPILE_FLAGS -Wall -Wextra -Weffc++)
+else()
+    set(COMPILE_FLAGS /W4)
+endif()
```

```suggestion
if (UNIX)
    add_compile_options(-Wall -Wextra -Weffc++)
else ()
    add_compile_options(/W4)
endif ()
```
You can do this instead

---

Using `target_` cmake functions is a good practice as it only edits the targets compile options. `add_compile_options` will add theses options to ALL targets in this directory and below (cf: add_subdirectory)